packages/backend-defaults/src/entrypoints/cache/CacheManager.ts

```diff
@@ -109,42 +106,200 @@ export class CacheManager {
     );
   }
 
-  /**
-   * Parse store-specific options from configuration.
-   *
-   * @param store - The cache store type ('redis', 'valkey', 'memcache', or 'memory')
-   * @param config - The configuration service
-   * @param logger - Optional logger for warnings
-   * @returns The parsed store options
-   */
   private static parseStoreOptions(
     store: string,
     config: RootConfigService,
     logger?: LoggerService,
   ): CacheStoreOptions | undefined {
     const storeConfigPath = `backend.cache.${store}`;
+    if (!config.has(storeConfigPath)) {
+      if (
+        store !== 'memory' &&
+        store !== 'memcache' &&
+        store !== 'redis' &&
+        store !== 'valkey' &&
+        store !== 'infinispan'
+      ) {
+        logger?.warn(
+          `No configuration found for cache store '${store}' at '${storeConfigPath}'.`,
+        );
+      }
+      return undefined;
```

Comment:
this is new - do you really want this mega-if at the very top that collects many behaviors in one place? if you want this check it should be inside the individual blocks below instead; only each block knows whether it's ok to not have a dedicated config section or not